### PR TITLE
Cards metadata - requirements [part of #1605]

### PR DESCRIPTION
**Pull request by chosta**
A chunk (hopefully good one) from the infamous #1605 that should be mergeworthy
+ CardRequirements remastered with builder pattern
+ VP, Description and Content Vue components
+ Some cards using the new new metadata requirements
+ sadly a bit of formatting leftover
+ new unit tests

Thanks Mr K9 for upping my TS game!

Plaese, review after 52ae7ca

### CODE REVIEW

**Comment by chosta** [comment]
Consider reviewing only after 
52ae7ca

File: src/cards/IProjectCard.ts [hunk 1]
```diff
@@ -13,5 +13,5 @@ export interface IProjectCard extends ICard {
     resourceType?: ResourceType;
     hasRequirements?: boolean;
     bonusResource?: Resources | undefined;
-    bonusMc?: number | undefined;
+    bonusMc?: number | undefined;
```
Comment (kberg) [inline src/cards/IProjectCard.ts:16]: This looks like a no-op change. But OK.

File: src/cards/venusNext/LuxuryFoods.ts [hunk 2]
```diff
@@ -11,14 +13,23 @@ export class LuxuryFoods implements IProjectCard {
...
     }
-    
+
     public getVictoryPoints() {
-      return 2;
+        return 2;
     }
+
+    public metadata: CardMetadata = {
+        description: "Requires that you have a Varth tag, an Earth tag and a Jovian tag.",
```
Comment (kberg) [inline src/cards/venusNext/LuxuryFoods.ts:28]: Varth Tag?

**Review by kberg — CHANGES_REQUESTED** [review]
One typo.

**Review by kberg — APPROVED** [review]
Fixed the typo.

**Comment by kberg** [comment]
Much, much easier. Thank you.l

**Pull request merged**